experimental/features_example_test.go

```diff
@@ -0,0 +1,166 @@
+package experimental_test
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"log"
+	"runtime"
+	"sync/atomic"
+
+	wazero "github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/experimental"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+)
+
+// pthreadWasm was generated by the following:
+//
+//	docker run -it --rm -v `pwd`/testdata:/workspace ghcr.io/webassembly/wasi-sdk:wasi-sdk-20 sh -c '$CC -o /workspace/pthread.wasm /workspace/pthread.c --target=wasm32-wasi-threads --sysroot=/wasi-sysroot -pthread -mexec-model=reactor -Wl,--export=run -Wl,--export=get'
+//
+// TODO: Use zig cc instead of wasi-sdk to compile when it supports wasm32-wasi-threads
+// https://github.com/ziglang/zig/issues/15484
```

Comment:
would it be possible now? Zig 0.11.0 seems to have it enabled https://github.com/ziglang/zig/commit/622c5f3200f66e9db74d696b5dbec308f61d506e

Reply:
Hmm, I think it might be supported for `zig` but not `zig cc`. Or do you see any other flag I need to try?

```
❯ zig cc -pthread -mexec-model=reactor -Wl,--export=run -Wl,--export=get --target=wasm32-wasi -o foo.wasm experimental/testdata/pthread.c
LLD Link... wasm-ld: error: /Users/anuraag/.cache/zig/o/a60f64a459c01f2581b65734bf5f66dd/pthread.o: undefined symbol: pthread_mutex_lock
wasm-ld: error: /Users/anuraag/.cache/zig/o/a60f64a459c01f2581b65734bf5f66dd/pthread.o: undefined symbol: pthread_mutex_unlock
wasm-ld: error: /Users/anuraag/.cache/zig/o/a60f64a459c01f2581b65734bf5f66dd/pthread.o: undefined symbol: pthread_mutex_lock
wasm-ld: error: /Users/anuraag/.cache/zig/o/a60f64a459c01f2581b65734bf5f66dd/pthread.o: undefined symbol: pthread_mutex_unlock
❯ zig cc -pthread -mexec-model=reactor -Wl,--export=run -Wl,--export=get --target=wasm32-wasi-threads -o foo.wasm experimental/testdata/pthread.c
error: UnknownApplicationBinaryInterface
❯ zig version
0.11.0
```

Reply:
ah ic.